config/config-mp/src/main/java/io/helidon/config/mp/MpConfigImpl.java

```diff
@@ -107,11 +106,15 @@
         });
     }
 
-    // TODO 3.0.0-JAKARTA
     @Override
-    public ConfigValue getConfigValue(String s) {
-        // TODO requires implementation of latest config version
-        return null;
+    public ConfigValue getConfigValue(String key) {
+        if (configProfile == null) {
+            return findConfigValue(key)
+                    .orElseGet(() -> new ConfigValueImpl(key, null, null, null, 0));
+        }
+        return findConfigValue("%" + configProfile + "." + key)
```

Comment:
Is the need for all this String concatenation required by the spec? Seems like not the fastest option

Reply:
The specification for profile requires that you first attempt to find the configuration value by the config profile prefix (which implies string concatenation), and then (if not found) you use the original key.

Reply:
Caching is not an option, as MP Config requires the values to be always requested from config source(s)